x-pack/plugins/security_solution/public/management/components/endpoint_responder/components/header_info/endpoint/header_endpoint_info.tsx

```diff
@@ -0,0 +1,51 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React, { memo } from 'react';
+import { EuiSkeletonText } from '@elastic/eui';
+import { EndpointAgentStatus } from '../../../../../../common/components/endpoint/endpoint_agent_status';
+import { HeaderAgentInfo } from '../header_agent_info';
+import { useGetEndpointDetails } from '../../../../../hooks';
+import { useGetEndpointPendingActionsSummary } from '../../../../../hooks/response_actions/use_get_endpoint_pending_actions_summary';
+import type { Platform } from '../platforms';
+
+interface HeaderEndpointInfoProps {
+  endpointId: string;
+}
+
+export const HeaderEndpointInfo = memo<HeaderEndpointInfoProps>(({ endpointId }) => {
+  const { data: endpointDetails, isFetching } = useGetEndpointDetails(endpointId, {
+    refetchInterval: 10000,
+  });
+  // fetch pending actions using the agent id and action status API
+  const { data: endpointPendingActions } = useGetEndpointPendingActionsSummary([endpointId], {
+    refetchInterval: 10000,
+  });
+
+  if (isFetching && endpointPendingActions === undefined) {
+    return <EuiSkeletonText lines={2} />;
+  }
```

Comment:
Will this cause a UI "flashing" condition when it refreshes (refetchInteval)?

The Skeleton lines should only show during initial load (so maybe use `isLoading` instead)